app/models/cross_model_slug.rb

```diff
@@ -0,0 +1,37 @@
+#
+# We have a simple top-level route equivalent to `/slug`. Because of this,
+# we want to verify that newly created records don't overlap with a previously-
+# defined `slug` -- in other words, slug values should be unique across all
+# relevant models. Except also, models might use "username" instead of "slug".
+#
+# Models can register with this class to be included in a cross-model-uniqueness
```

Comment:
I suppose we should remove "register" terminology from the comments as registration was removed.

Reply:
Tried to re-word this in d4f29305f